### PR TITLE
Handle project having experiments with no data

### PR DESCRIPTION
**Pull request by mattseddon**
Fixes: 

```
When I'm trying to show experiments/plots, that "You project contains no data".
But I have some... experiments and those plots
Any ideas why is VSCode behave like this?  Also, no clear what should I do here
```

From https://iterativeai.slack.com/archives/CB41NAL8H/p1701257451702629

### CODE REVIEW

File: extension/src/experiments/webview/messages.ts [hunk 1]
```diff
@@ -85,7 +85,10 @@ export class WebviewMessages {
     const data = await this.getWebviewData()
 
     const hasNoRows = data.rows.length === 0
-    const hasNoData = !this.columns.hasNonDefaultColumns() || hasNoRows
+    const hasNoExperiments =
+      this.experiments.getExperimentsAndQueued().length === 0
+    const hasNoData =
+      (!this.columns.hasNonDefaultColumns() && hasNoExperiments) || hasNoRows
```
Comment (mattseddon) [inline extension/src/experiments/webview/messages.ts:91]: [I] Need to handle the tree as well, make an `ExperimentsModel` helper.

**Review by julieg18 — COMMENTED** [review]
This updates the experiments tree but the webview is still showing the "has no data" message:

Screenshot 2023-11-30 at 8 19 43 AM

I updated `Experiments.getHasData()` to see if the webviews could work without none default columns but no luck: 

Screenshot 2023-11-30 at 8 18 24 AM

**Comment by mattseddon** [comment]
> This updates the experiments tree but the webview is still showing the "has no data" message:

@julieg18 what data/setup are you using?

**Review by julieg18 — APPROVED** [review]
Single plots project is working now, great work!

**Pull request merged**